components/DataActiveView/index.js

```diff
@@ -0,0 +1,54 @@
+import PropTypes from 'prop-types';
+import React, {useState, useMemo, useCallback} from 'react';
+import {View, Text, TouchableOpacity} from 'react-native';
+import CheckIcon from '../../assets/icons/Check.svg';
+import {horizontalScale, verticalScale} from '../../assets/metrics/Metrics';
+import CreditCardDeletion from '../../screens/AccountSettingsScreen/CreditCardDeletion';
+import styles from './style';
+
+const DataActiveView = ({ item, isCreditCard, id }) => {
+    const [isModalVisible, setModalVisible] = useState(false);
+
+    const value = useMemo(() => (isCreditCard ? `••••${item}` : item), [isCreditCard, item]);
+
+    const handleDeleteCard = useCallback(() => {
+        setModalVisible(true);
+    }, []);
+
+    const selectAppropriateStyle = (className) => ([
+        styles[className],
+        { left: isCreditCard ? horizontalScale(40) : 0 },
+    ]);
+
+    if (!item) {
+        return null;
+    }
+
+    return (
+        <View style={styles.container}>
+            <Text style={selectAppropriateStyle('value', isCreditCard)}>
+                {value}
+            </Text>
+            <View style={styles.checked}>
+                {isCreditCard ? (
+                    <TouchableOpacity style={styles.checkedRemove} onPress={handleDeleteCard}>
```

Comment:
<TouchableOpacity style={styles.checkedRemove} onPress={handleDeleteCard}>
                        <Text style={styles.removeCard}>✖️</Text>
</TouchableOpacity>
  this code is repeated, move to separate component